src/libultra/io/contramread.c

```diff
@@ -1,3 +1,63 @@
 #include "global.h"
 
-#pragma GLOBAL_ASM("asm/non_matchings/boot/contramread/__osContRamRead.s")
+#define BLOCKSIZE 32
+
+s32 __osPfsLastChannel = -1;
+
+s32 __osContRamRead(OSMesgQueue* ctrlrqueue, s32 channel, u16 addr, u8* data) {
```

Comment:
Any reason why use `u8*` for `data` instead of `void*`?

Reply:
It's used for a crc thing in the other file, which takes a byte array, so I'll stick with u8 and rename it to buffer.